src/cairoUtilFuncGen/memory/arrayConcat.ts

```diff
@@ -0,0 +1,147 @@
+import { DataLocation, FunctionCall, getNodeType, PointerType, TypeName } from 'solc-typed-ast';
+import { printNode, printTypeNode } from '../../utils/astPrinter';
+import { NotSupportedYetError, TranspileFailedError } from '../../utils/errors';
+import { createCairoFunctionStub, createCallToFunction } from '../../utils/functionGeneration';
+import { mapRange, typeNameFromTypeNode } from '../../utils/utils';
+import { uint256 } from '../../warplib/utils';
+import { CairoFunction, StringIndexedFuncGen } from '../base';
+
+const IMPLICITS = '{range_check_ptr, warp_memory : DictAccess*}';
+
+export class MemoryArrayConcat extends StringIndexedFuncGen {
+  gen(concat: FunctionCall) {
+    const args = concat.vArguments;
+    args.forEach((expr) => {
+      const exprType = getNodeType(expr, this.ast.compilerVersion);
+      if (!(exprType instanceof PointerType))
+        throw new NotSupportedYetError(
+          `Concatenation of expression ${printNode(expr)} of type ${printTypeNode(
+            exprType,
+          )} not supported yet`,
+        );
+      if (exprType.location !== DataLocation.Memory)
+        throw new TranspileFailedError(
+          `Concatenation of '${exprType.location}' different than memory`,
```

Comment:
Since DataAccessFunctionaliser runs after this I don't think the location should be specified here. Instead we should rely on the appropriate copy from storage or calldata getting inserted if necessary. Could you also then add these cases to the tests